static/app/components/devtoolbar/hooks/useReplayRecorder.tsx

```diff
@@ -0,0 +1,118 @@
+import {useCallback, useEffect, useState} from 'react';
+import type {replayIntegration} from '@sentry/react';
+import type {ReplayRecordingMode} from '@sentry/types';
+
+import useConfiguration from 'sentry/components/devtoolbar/hooks/useConfiguration';
+
+type ReplayRecorderState = {
+  disabledReason: string | undefined;
+  isDisabled: boolean;
+  isRecording: boolean;
+  lastReplayId: string | undefined;
+  start(): Promise<boolean>;
+  stop(): Promise<boolean>;
+};
+
+function getIsRecording(
+  replay: ReturnType<typeof replayIntegration> | undefined
+): boolean {
+  return replay?._replay.isEnabled() ?? false;
```

Comment:
probably want one helper `getReplayInternal(replay: ReturnType<typeof replayIntegration> | undefined) { return replay?._replay; }` instead of these three.

The purpose of that single helper would be so there's only one line that needs the lint error silenced.

Reply:
gotcha, that's the type of the public api though, we probably don't have access to the private type